content/en/docs/reference/using-api/deprecation-guide.md

```diff
@@ -0,0 +1,265 @@
+
+#### Deployment
+
+The **extensions/v1beta1**, **apps/v1beta1**, and **apps/v1beta2** API versions of Deployment are no longer served as of v1.16.
+
+* Migrate manifests and API clients to use the **apps/v1** API version, available since v1.9.
+* All existing persisted objects are accessible via the new API
+* Notable changes:
+    * `spec.rollbackTo` is removed
+    * `spec.selector` is now required and immutable after creation; use the existing template labels as the selector for seamless upgrades
+    * `spec.progressDeadlineSeconds` now defaults to `600` seconds (the default in `extensions/v1beta1` was no deadline)
+    * `spec.revisionHistoryLimit` now defaults to `10` (the default in `apps/v1beta1` was `2`, the default in `extensions/v1beta1` was to retain all)
+    * `maxSurge` and `maxUnavailable` now default to `25%` (the default in `extensions/v1beta1` was `1`)
+
+#### StatefulSet
+
+The **apps/v1beta1** and **apps/v1beta2** API versions of StatefulSet are no longer served as of v1.16.
+
+* Migrate manifests and API clients to use the **apps/v1** API version, available since v1.9.
+* All existing persisted objects are accessible via the new API
+* Notable changes:
+    * `spec.selector` is now required and immutable after creation; use the existing template labels as the selector for seamless upgrades
+    * `spec.updateStrategy.type` now defaults to `RollingUpdate` (the default in `apps/v1beta1` was `OnDelete`)
+
+#### ReplicaSet
+
+The **extensions/v1beta1**, **apps/v1beta1**, and **apps/v1beta2** API versions of ReplicaSet are no longer served as of v1.16.
+
+* Migrate manifests and API clients to use the **apps/v1** API version, available since v1.9.
+* All existing persisted objects are accessible via the new API
+* Notable changes:
+    * `spec.selector` is now required and immutable after creation; use the existing template labels as the selector for seamless upgrades
+
+## What To Do
+
+Kubernetes 1.22 will be released later in 2021, so be sure to audit
+your configuration and integrations now!
+
+### Test with deprecated APIs disabled
+
+You can test your clusters by starting an API server with specific API versions disabled
+to simulate upcoming removals. Add the following flag to the API server startup arguments:
+
+`--runtime-config=$group/$version=false`
+
+For example:
+
+`--runtime-config=admissionregistration.k8s.io/v1beta1=false,apiextensions.k8s.io/v1beta1,...`
+
+### Locate use of deprecated APIs
+
+Use [client warnings, metrics, and audit information available in 1.19+](https://kubernetes.io/blog/2020/09/03/warnings/#deprecation-warnings)
+to locate use of deprecated APIs.
+
+### Migrate to non-deprecated APIs
+
+* Update custom integrations and controllers to call the non-deprecated APIs
+* Change YAML files to reference the non-deprecated APIs
+
+    You can use the `kubectl-convert` command (`kubectl convert` prior to v1.20)
```

Comment:
How do people install `kubectl convert`? I suspect they will want to know (and I don't think we document it).

Reply:
(it's OK, IMO, to log an issue about documenting it and making it clear there's an open issue on that topic - maybe we mark it insta-frozen so it doesn't get overlooked).

Reply:
`kubectl convert` prior to 1.20 doesn't need installing. https://github.com/kubernetes/kubernetes/issues/99076 tracks making a pre-built `kubectl-convert` easily available/installable.

Reply:
We never logged the docs issue as far as I know. This came up: https://github.com/kubernetes/website/issues/38809